packages/loader/container-utils/src/error.ts

```diff
@@ -77,32 +79,43 @@ export class DataCorruptionError extends LoggingError implements IErrorBase {
     }
 }
 
-export class DataProcessingError extends LoggingError implements IErrorBase {
+export class DataProcessingError extends LoggingError implements IErrorBase, IFluidErrorBase {
     readonly errorType = ContainerErrorType.dataProcessingError;
     readonly canRetry = false;
 
-    constructor(errorMessage: string, props?: ITelemetryProperties) {
+    constructor(
+        errorMessage: string,
+        readonly fluidErrorCode: string,
+        props?: ITelemetryProperties,
+    ) {
         super(errorMessage, props);
     }
 
     /**
      * Conditionally coerce the throwable input into a DataProcessingError.
      * @param originalError - Throwable input to be converted.
      * @param message - Sequenced message (op) to include info about via telemetry props
+     * @param errorCodeIfNone - errorCode identifying the call site, to be used if the originalError has no error code.
      * @returns Either a new DataProcessingError, or (if wrapping is deemed unnecessary) the given error
      */
     static wrapIfUnrecognized(
         originalError: any,
+        errorCodeIfNone: string,
         message: ISequencedDocumentMessage | undefined,
-    ): ICriticalContainerError {
-        const newErrorFn = (errMsg: string) => new DataProcessingError(errMsg);
+    ): IFluidErrorBase {
+        const newErrorFn = (errMsg: string) => {
+            const dpe = new DataProcessingError(errMsg, errorCodeIfNone);
+            dpe.addTelemetryProperties({ untrustedOrigin: true}); // To match normalizeError
```

Comment:
doesn't it set it to `1` not `true` will those be the same in telemetry?

Reply:
Ok yeah I'll make it 1 here to match, good point.

Reply:
```suggestion
            dpe.addTelemetryProperties({ untrustedOrigin: 1}); // To match normalizeError
```